src/Language/Haskell/Brittany/Internal/Backend.hs

```diff
@@ -173,6 +175,8 @@ layoutBriDocM = \case
                       -- evil hack for CPP:
                       case comment of
                         ('#':_) -> layoutMoveToCommentPos y (-999)
+                        "("     -> pure ()
+                        ")"     -> layoutMoveToCommentPosX (x - 1)
```

Comment:
It was for the case where you have:
```
type ((MyType a) b) = (a, b)
```
to avoid it layouting as
```
type ((MyType a ) b ) = (a, b)
```

You're right though it's hacky, makes the backend more complicated, and assumes the thing before the ")" is a `docSeparator`. Without it we would need each type variable to know whether the next one has a comment or not. I've thought about alternatives a little, so let me know if you'd like me to do it another way or if this doesn't make sense!

Reply:
This behaviour is only on `8.4.3` where the annotations are moved into comments.